src/dmd/parse.d

```diff
@@ -6602,9 +6602,13 @@ final class Parser(AST) : Lexer
                     {
                         nextToken();
                         AST.ExpInitializer expInit = value.isExpInitializer();
-                        assert(expInit);
-                        e = expInit.exp;
-                        value = parseInitializer();
+                        if (!expInit)
+                            error("initialzer expected following colon, not `%s`", token.toChars());
```

Comment:
initialzer -> initializer expression

Reply:
Thanks, fixed.

Reply:
I actually meant to also add "expression", too, because there are other types of initializers, and the user might be confused if he has supplied a valid array initializer.

Reply:
Done.